torch/utils/_pytree.py

```diff
@@ -180,11 +180,10 @@ class _SerializeNodeDef(NamedTuple):
 
     _cxx_pytree_dynamo_traceable = Version(_optree_version) >= Version("0.13.0")
     if not _cxx_pytree_dynamo_traceable:
-        warnings.warn(
+        raise RuntimeError(
             "optree is installed but the version is too old to support PyTorch Dynamo in C++ pytree. "
             "C++ pytree support is disabled. "
-            "Please consider upgrading optree using `python3 -m pip install --upgrade 'optree>=0.13.0'`.",
-            FutureWarning,
+            "Please consider upgrading optree using `python3 -m pip install --upgrade 'optree>=0.13.0'`."
```

Comment:
```suggestion
            "Please upgrade optree using `python3 -m pip install --upgrade 'optree>=0.13.0'`."
```
Can be direct giving the solution, considering will not overcome the problem ;)

Reply:
I'd also future proof this by not manually hardcoding the version in the assert but using a single version variable earlier and use it in 2 places. 

otherwise often one place is updated but not the other and problems ensue.

```
optree_min_version = "0.13.0"
_cxx_pytree_dynamo_traceable = Version(_optree_version) >= Version(optree_min_version)
...
f"Please upgrade optree using `python3 -m pip install --upgrade 'optree>={optree_min_version}'`."
```